### PR TITLE
feat!: Remove simple::SimpleProtocol

### PR DESCRIPTION
**Pull request by mxinden**
## Description

Due to lack of usage, remove the `simple::SimpleProtocol`. Users may look at `libp2p::core::upgrade` for alternatives.

## Notes

Discovered through @umgefahren's `master` docs (https://github.com/libp2p/rust-libp2p/pull/3126).

## Open Questions

- @thomaseizinger what would be the right conventional commit prefix here?

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,12 @@
 
 # `libp2p` facade crate
 
+# 0.51.0 [unreleased]
+
+- Remove `SimpleProtocol` due to being unused. See `libp2p::core::upgrade` for alternatives. See [PR 3191].
```
Comment (jxs) [inline CHANGELOG.md]:
```suggestion
- Remove `SimpleProtocol` due to being unused. See [`libp2p::core::upgrade`](https://docs.rs/libp2p/0.50.0/libp2p/core/upgrade/index.html) for alternatives. See [PR 3191].
```

**Review by jxs — APPROVED** [review]
LGTM

**Comment by thomaseizinger** [comment]
> @thomaseizinger what would be the right conventional commit prefix here?

Uhm, good question. This [discussion](https://stackoverflow.com/questions/48075169/semantic-commit-type-when-remove-something) discussion suggests `refactor!` or `feat!`. Personally I tend towards `feat!` because refactoring something shouldn't change the feature set.

**Review by thomaseizinger — APPROVED** [review]

**Comment by mxinden** [comment]
Testground failure is due to a rename:

```
failed to load composition file: failed to process composition template: open ping/_compositions/go-rust-interop-latest.toml: no such file or directory
```

https://github.com/libp2p/test-plans/pull/70

See https://github.com/libp2p/rust-libp2p/pull/2647#issuecomment-1335189994 for `libp2p-relay` and `libp2p-dcutr` failures.

**Comment by thomaseizinger** [comment]
> Testground failure is due to a rename:
> 
> ```
> failed to load composition file: failed to process composition template: open ping/_compositions/go-rust-interop-latest.toml: no such file or directory
> ```
> 
> https://github.com/libp2p/test-plans/pull/70

Fix is here: https://github.com/libp2p/rust-libp2p/pull/3203

**Comment by mxinden** [comment]
For `libp2p-dcutr` and `libp2p-relay` failure, see https://github.com/libp2p/rust-libp2p/pull/2647#issuecomment-1335189994.

**Pull request merged**